backend/src/search/meta.rs

```diff
@@ -0,0 +1,58 @@
+use meilisearch_sdk::indexes::Index;
+use serde::{Serialize, Deserialize};
+
+use crate::prelude::*;
+
+use super::VERSION;
+
+
+/// Data stored in the 'meta' index.
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct Meta {
+    pub(crate) id: String,
+
+    /// A number simply counting up whenever we change the search index in a way
+    /// that requires a rebuild.
+    pub(crate) version: u32,
+
+    /// A flag that is set to `true` while transitioning to a new index version.
+    /// If the transition fails or is interrupted, the flag stays set and
+    /// Tobira will rebuild again next time.
+    pub(crate) dirty: bool,
```

Comment:
Naive question: Why do we need this? Why can't we just update the version at the very end of the process?

Reply:
Good question. So my thinking was: what if something breaks while we are rebuilding. Then we would still have the old version in the meta table, but everything else already partially rebuild. Can that happen? Mhh I guess it technically could (e.g. due to communication issues with Meili). If for example, we only delete one index, but the others remain. Or if we only manage to populate one. In both cases the whole index is somewhat in a broken state. If someone were then to rollback to an older Tobira version that's happy with the previous search index schema... there wont be any rebuilding, but stuff would be broken.

So I guess it's rather unusual that an actual problem would occur. But the dirty flag at least protects from some stuff, so it's good? Maybe we should rather set the `dirty` flag to true in the `rebuild` command as well.

Reply:
The rollback case is a fair point. :thinking: Maybe we could add a comment to that effect?